easybuild/easyblocks/g/gromacs.py

```diff
@@ -372,55 +453,19 @@ def install_step(self):
-            if LooseVersion(self.version) < LooseVersion('4.6'):
-
-                cmd = "make distclean"
-                (out, _) = run_cmd(cmd, log_all=True, simple=False)
-
-                self.cfg.update('configopts', "--enable-mpi --program-suffix={0}".format(self.cfg['mpisuffix']))
-                ConfigureMake.configure_step(self)
-
-                super(EB_GROMACS, self).build_step()
-
-                super(EB_GROMACS, self).install_step()
-
-            else:
-                self.cfg['configopts'] = re.sub(r'-DGMX_MPI=OFF', r'', self.cfg['configopts'])
-
-                if self.cfg['mpi_numprocs'] == 0:
-                    self.log.info("No number of test MPI tasks specified -- using default: %s" % self.cfg['parallel'])
-                    self.cfg['mpi_numprocs'] = self.cfg['parallel']
-
-                elif self.cfg['mpi_numprocs'] > self.cfg['parallel']:
-                    self.log.warning("Number of test MPI tasks (%s) is greater than value for 'parallel': %s",
-                                     self.cfg['mpi_numprocs'], self.cfg['parallel'])
-
-                self.cfg.update('configopts', "-DGMX_MPI=ON -DGMX_THREAD_MPI=OFF")
-
-                mpiexec = which(self.cfg['mpiexec'])
-                if mpiexec:
-                    self.cfg.update('configopts', "-DMPIEXEC=%s" % mpiexec)
-                    self.cfg.update('configopts', "-DMPIEXEC_NUMPROC_FLAG=%s" % self.cfg['mpiexec_numproc_flag'])
-                    self.cfg.update('configopts', "-DNUMPROC=%s" % self.cfg['mpi_numprocs'])
-                elif self.cfg['runtest']:
-                    raise EasyBuildError("'%s' not found in $PATH", self.cfg['mpiexec'])
-
-
-                self.log.info("Using %s as MPI executable when testing, with numprocs flag '%s' and %s tasks",
-                              self.cfg['mpiexec'], self.cfg['mpiexec_numproc_flag'], self.cfg['mpi_numprocs'])
+        # Reset installopts etc for the benefit of the gmxapi extension
+        self.cfg['installopts'] = self.save_installopts
 
-                # clean up obj dir before reconfiguring
-                shutil.rmtree(os.path.join(self.builddir, 'easybuild_obj'))
+    def extensions_step(self, fetch=False):
+        """ Custom extensions step, only handle extensions after the last iteration round"""
+        if self.iter_idx < self.variants_to_build-1:
+            self.log.info("skipping extension step %s", self.iter_idx)
+            return
 
-                # rebuild/test/install with MPI options
-                super(EB_GROMACS, self).configure_step()
-                super(EB_GROMACS, self).build_step()
-                super(EB_GROMACS, self).test_step()
-                super(EB_GROMACS, self).install_step()
-
-                self.log.info("A full regression test suite is available from the GROMACS web site")
+        # Set runtest to None so that the gmxapi extension doesn't try to
+        # run "check" as a command
+        self.cfg['runtest'] = None
```

Comment:
Shouldn't you restore the original `runtest` value afterwards, since a custom value may have been specified in the easyconfig file?

Reply:
Do I need to? This is at the last iteration and extensions_step is that last thing done. So nothing else should happen as far as I know.

Reply:
Anyway, done